query-engine/connectors/mongodb-query-connector/src/interface/connection.rs

```diff
@@ -79,6 +79,17 @@ impl WriteOperations for MongoDbConnection {
         .await
     }
 
+    async fn create_records_returning(
+        &mut self,
+        _model: &Model,
+        _args: Vec<WriteArgs>,
+        _skip_duplicates: bool,
+        _selected_fields: FieldSelection,
+        _trace_id: Option<String>,
+    ) -> connector_interface::Result<ManyRecords> {
+        unimplemented!()
+    }
+
```

Comment:
In general, having trait implementations that are `unimplemented!()`  feel wrong. We have a few traits in the codebase which purpose is to describe a grouping of functionality (like WriteOperations) rather than a contract, and might be segregated into smaller traits. 

Unfortunately, doing so, will mean that wherever we depend on more than of the segregated contracts, either we would need to define a supertrait, or repeat the union of traits.

This is related to https://github.com/prisma/prisma-engines/pull/4698/files#r1499449038, a single `create_records` with more responsibility will have an implementation in every connector. Still, this does not change the direction of that comment. I won't prefer piggibacking functionality into the same function, if it can be segregated.

Reply:
I have the same feeling about `unimplemented!` here, but also agree that the alternative (merging `create_records` and `create_records_returning` into one function) is worse. I will leave it as it is for now. We can refactor later if we want to, as you said.